src/VisualStudio/CSharp/Impl/LanguageService/CSharpLanguageService_ICSharpProjectHost.cs

```diff
@@ -16,6 +17,15 @@ internal partial class CSharpLanguageService : ICSharpProjectHost
         public void BindToProject(ICSharpProjectRoot projectRoot, IVsHierarchy hierarchy)
         {
             var projectName = Path.GetFileName(projectRoot.GetFullProjectName()); // GetFullProjectName returns the path to the project file w/o the extension?
+            var projectFilename = AbstractLegacyProject.GetProjectFilePath(hierarchy);
+            var projectId = Workspace.ProjectTracker.GetOrCreateProjectIdForPath(projectFilename, projectName);
+
+            if (Workspace.Services.GetService<IDeferredProjectWorkspaceService>()?.IsDeferredProjectLoadEnabled ?? false)
```

Comment:
Will fix style, comment, and name issues if this is not approved today.  Otherwise, will fix in master.